tests/integration/test_authd/test_authd_use_password_invalid.py

```diff
@@ -0,0 +1,150 @@
+'''
+copyright: Copyright (C) 2015-2023, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check invalid values in the authd.pass (for now just checks 'empty')
+       raises the expected error logs.
+
+components:
+    - authd
+
+suite: use_password
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-authd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+tags:
+    - enrollment
+    - authd
+'''
+
+import pytest
+
+from pathlib import Path
+
+from wazuh_testing.modules.authd import event_monitor as evm
+from wazuh_testing.tools import DEFAUL_AUTHD_PASS_PATH
+from wazuh_testing.tools.file import write_file, delete_file
+from wazuh_testing.tools.configuration import get_test_cases_data, load_configuration_template
+from wazuh_testing.tools.services import control_service
+
+
+# Constants & base paths
+TEST_NAME = Path(__file__).stem.replace('test_', '')
+DATA_PATH = Path(Path(__file__).parent, 'data')
+TESTS_CASES_PATH = Path(DATA_PATH, 'test_cases')
+CONFIGS_PATH = Path(DATA_PATH, 'config_templates')
+
+# Marks
+pytestmark = [pytest.mark.server, pytest.mark.tier(level=1)]
+
+# Paths
+cases_path = Path(TESTS_CASES_PATH, f"cases_{TEST_NAME}.yaml")
+config_path = Path(CONFIGS_PATH, f"config_{TEST_NAME}.yaml")
+
+# Configurations and test cases
+params, metadata, case_ids = get_test_cases_data(cases_path)
+configuration = load_configuration_template(config_path, params, metadata)
+local_internal_options = {'authd.debug': '2'}
+
+
+# Fixture
+@pytest.fixture()
+def set_authd_pass(metadata: dict):
+    """Configure the file 'authd.pass' as needed for the test."""
+    # Write the content in the authd.pass file.
+    write_file(DEFAUL_AUTHD_PASS_PATH, metadata.get('password'))
+
+    yield
+
+    # Delete the file as by default it doesn't exist.
+    delete_file(DEFAUL_AUTHD_PASS_PATH)
+
```

Comment:
Why reason you add it here, and not in `modules/authd/init.py` or `modules/authd/utils.py`?

Reply:
Because some other test files could have a similar fixture maybe more generic, it will grab the password from the metadata key "password" making this fixture more specific for this test, if all the other test files has the same keys in the yaml it could be included in a `conftest.py` but for what I saw, every test file there has different yamls keys to do similar things